Adding show_completed argument to iCall

sanchezfauste opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
Adding show_completed argument to iCall to be able to hide the completed Meetings, Calls and Tasks.

## Description
<!--- Describe your changes in detail -->
Adding a parameter to specify if the completed tasks, meetings and calls have to be present to the returned icall. By default this parameter is true.

<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here unless your commit contains the issue number -->

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
We think that showing the completed tasks in the calendar is not so useful when there are a lot of tasks. It would be very useful if you can choose if you want to see the completed tasks or not.

## How To Test This
<!--- Please describe in detail how to test your changes. -->
Obtain iCall without completed tasks, meetings and calls: https://crm.example.com/ical_server.php?type=ics&key=XXX&email=YYY&show_tasks_as_events=true&show_completed=false

Obtain iCall including completed tasks, meetings and calls: https://crm.example.com/ical_server.php?type=ics&key=XXX&email=YYY&show_tasks_as_events=true&show_completed=true

or simply:

https://crm.example.com/ical_server.php?type=ics&key=XXX&email=YYY&show_tasks_as_events=true

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist
<!--- Go over all the following points and check all the boxes that apply. --->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! --->
- [x] My code follows the code style of this project found [here](https://suitecrm.com/wiki/index.php/Coding_Standards).
- [x] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM) guidelines.

<!--- Your pull request will be tested via Travis CI to automatically indicate that your changes do not prevent compilation. --->

<!--- If it reports back that there are problems, you can log into the Travis system and check the log report for your pull request to see what the problem was. --->

sanchezfauste committed:
Adding show_completed argument to iCall

Dillon-Brown commented:
Hi @sanchezfauste You still need to sign the Clahub Contributor License Agreement.

sanchezfauste commented:
Thank you @Dillon-Brown, I have signed the Contributor License.

isleshocky77 committed:
Adding the abily to relate a Note with a Campaign

isleshocky77 committed:
PSR-2 Convert tabs to spaces

isleshocky77 committed:
Code Format after tab -> spaces conversion

isleshocky77 committed:
Remove duplicate array key (clean warning)

Dillon-Brown committed:
Remove fullcalendar

Dillon-Brown committed:
Fixed #3057 - FullCalendar update to version 3.1

Dillon-Brown committed:
Fixed #4511 - Upgrade of TinyMCE to latest 4.7.x version

Dillon-Brown committed:
Remove tinyMCE

Dillon-Brown committed:
Fixed #3100 - Upgrade for HTML Purifier [Request]

Dillon-Brown committed:
Merge pull request #6714 from gymad/develop_elasticpager_fix

fix last page when modulus is zero

Dillon-Brown committed:
Merge branch 'develop' into tiny_mce

Dillon-Brown committed:
Merge pull request #6730 from Dillon-Brown/issue-6720

Fixed #6720 - Remove old recaptcha files

Dillon-Brown committed:
Merge branch 'develop' into fullcalendar

Dillon-Brown committed:
Merge branch 'develop' into fullcalendar

isleshocky77 committed:
Cleaning up the clean cache command

* Update command to allow for cache directory specified in config
* Remove warnings/errors when no cached directory present
* Update to have namespace "cache" so we can add other cache commands later
  * Leave alias for current command
* Use provided toolsets required by robo for implementation
* Print cleaned directories
* Confirm before cleaning directory
  * Allow for force override

Dillon-Brown committed:
Merge branch 'develop' into fix-clean-cache-command-config-cache-dir

isleshocky77 committed:
Code formatting

Dillon-Brown committed:
Merge remote-tracking branch 'upstream/develop' into feature/campaign-notes

# Conflicts:
#	modules/Campaigns/Campaign.php
#	modules/Campaigns/views/view.detail.php
#	modules/Notes/vardefs.php

Dillon-Brown committed:
Reformat

Dillon-Brown committed:
Fix test

samus-aran committed:
Merge pull request #4606 from isleshocky77/feature/campaign-notes

Add ability to have Notes on a Campaign

Dillon-Brown committed:
Formatting

Dillon-Brown committed:
Merge pull request #6802 from isleshocky77/fix-clean-cache-command-config-cache-dir

Allow for clean cache command to use config'd cache_dir (plus clean up)

lazka committed:
Fixes #7111 - composer: include mpdf/mpdf 6.1 and drop in-tree version (5.7)

Remove the included version of mpdf (in modules/AOS_PDF_Templates/PDF_Lib) and
add it to composer instead.

This also updates the version from 5.7.1 to 6.1.0 to get some fixes that were already
applied manually to the version in SuiteCRM. There are many newer versions available,
but they require more changes and this is a good first step (and this potentially
could be backported to 7.10)

The version in tree had one manual fix applied (d1279cad9c717) which is included in
6.1.0 already, so the upstream version can be used as is.

The required changes to make things compatible with with 6.1 were
* mPDF_exception is now called MpdfException
* SetAutoFont() no longer exists, the docs suggest to use autoLangToFont instead
  to get the same results as before.

Fixes #7111

Dillon-Brown committed:
Merge remote-tracking branch 'upstream/develop' into fullcalendar

# Conflicts:
#	composer.json
#	composer.lock

Dillon-Brown committed:
Merge remote-tracking branch 'upstream/hotfix' into develop_merge

code-ph0y committed:
Merge pull request #6702 from Dillon-Brown/fullcalendar

Fixed #3057 - FullCalendar update to version 3.1

Dillon-Brown committed:
Merge branch 'develop' into develop_merge

samus-aran committed:
Merge pull request #6703 from Dillon-Brown/htmlpurifier

Fixed #3100 - Upgrade for HTML Purifier [Request]

Dillon-Brown committed:
Merge remote-tracking branch 'upstream/master' into develop_merge

Dillon-Brown committed:
Merge remote-tracking branch 'origin/develop_merge' into develop_merge

# Conflicts:
#	composer.json

Dillon-Brown committed:
Update composer.lock

Dillon-Brown committed:
Merge pull request #7131 from Dillon-Brown/develop_merge

Develop merge

Dillon-Brown committed:
Add pclzip/pclzip to composer

Dillon-Brown committed:
Remove pclzip references

Dillon-Brown committed:
Remove pclZip from include

Dillon-Brown committed:
Update composer.lock

Dillon-Brown committed:
Remove zip_utils.php

Dillon-Brown committed:
Reformat php_zip_utils.php

Dillon-Brown committed:
Update composer

Dillon-Brown committed:
Merge remote-tracking branch 'upstream/develop' into tiny_mce

# Conflicts:
#	composer.json
#	composer.lock

Dillon-Brown committed:
Update composer

Dillon-Brown committed:
Update tinymce references

Dillon-Brown committed:
Add tedivm/jshrink to composer

Dillon-Brown committed:
Remove Minifier.php

Dillon-Brown committed:
Fixed #7027 - Use a maintained library to minify JS

Dillon-Brown committed:
Merge remote-tracking branch 'upstream/develop' into fix-mpdf-composer

# Conflicts:
#	composer.json
#	composer.lock

Dillon-Brown committed:
Update composer

Dillon-Brown committed:
PHP 5.6 compat

code-ph0y committed:
Merge pull request #6699 from Dillon-Brown/tiny_mce

Fixed #4511 - Upgrade of TinyMCE to latest 4.9.x version

Dillon-Brown committed:
Merge branch 'develop' into fix-mpdf-composer

Dillon-Brown committed:
Merge pull request #7118 from lazka/fix-mpdf-composer

Fixes #7111 - composer: include mpdf/mpdf 6.1 and drop in-tree version (5.7)

code-ph0y committed:
Merge branch 'develop' into issue-7027

cameronblaikie committed:
Merge pull request #7295 from Dillon-Brown/issue-7027

Fixed #7027 - Use a maintained library to minify JS

Dillon-Brown committed:
Merge remote-tracking branch 'upstream/develop' into feature/plcZip

# Conflicts:
#	composer.json
#	composer.lock

Dillon-Brown committed:
Update composer.lock

Dillon-Brown committed:
Merge remote-tracking branch 'upstream/develop' into feature/plcZip

# Conflicts:
#	composer.lock

Dillon-Brown committed:
Update composer.lock

Dillon-Brown committed:
Merge pull request #7287 from Dillon-Brown/feature/plcZip

Remove PclZip

Mac-Rae commented:
Please resolve the conflicts on this pull request bringing it up to date.

sanchezfauste committed:
Merge branch 'develop' into ical_show_completed

sanchezfauste commented:
> Please resolve the conflicts on this pull request bringing it up to date.

Done.
